src/hooks/useDeviceSize.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const useMediaQuery = (query) => {
+  const [matches, setMatches] = useState(false);
+
+  useEffect(() => {
+    const media = window.matchMedia(query);
+    const updateMatch = () => setMatches(media.matches);
+    updateMatch();
+
+    media.addEventListener('change', updateMatch);
+    return () => media.removeEventListener('change', updateMatch);
+  }, [query]);
+
+  return matches;
+};
+
+const useDeviceSize = () => {
+  const isDesktop = useMediaQuery('(min-width: 1024px)');
+  const isTablet = useMediaQuery('(min-width:768px) and (max-width:1023px)');
+  const isMobile = useMediaQuery('(max-width: 767px)');
+
+  return { isDesktop, isTablet, isMobile };
```

Comment:
위와 같은 이유로 `useDeviceSize` 훅의 상태 값에도 `useMemo`를 사용할 수 있습니다.
```suggestion
  return useMemo(() => ({ isDesktop, isTablet, isMobile }), [isDesktop, isTablet, isMobile]);
```